lib/propolis-client/src/lib.rs

```diff
@@ -1,194 +1,116 @@
-//! Interface for making API requests to propolis.
-//! This should be replaced with a client generated from the OpenAPI spec.
+// Copyright 2022 Oxide Computer Company
+//! A client for the Propolis hypervisor frontend's server API.
+//!
+//! It is being experimentally migrated to `progenitor` for auto-generation,
+//! which is opt-in at present with crate feature `generated`, and additional
+//! compatibility impls and re-exports to approximate the former handmade
+//! bindings' module layout with crate feature `generated-migration`.
 
-use reqwest::Body;
-use reqwest::IntoUrl;
-use serde::de::DeserializeOwned;
-use slog::{info, o, Logger};
-use std::net::SocketAddr;
-use thiserror::Error;
-use uuid::Uuid;
-
-pub mod api;
 pub mod instance_spec;
 
-/// Errors which may be returend from the Propolis Client.
-#[derive(Debug, Error)]
-pub enum Error {
-    #[error("Request failed: {0}")]
-    Reqwest(#[from] reqwest::Error),
-
-    #[error("Bad Status: {0}")]
-    Status(u16),
-}
-
-/// Client-side connection to propolis.
-pub struct Client {
-    client: reqwest::Client,
-    log: Logger,
-    address: SocketAddr,
-}
-
-// Sends "request", awaits "response", and returns an error on any
-// non-success status code.
-//
-// TODO: Do we want to handle re-directs?
-async fn send_and_check_ok(
-    request: reqwest::RequestBuilder,
-) -> Result<reqwest::Response, Error> {
-    let response = request.send().await.map_err(Error::from)?;
-
-    if !response.status().is_success() {
-        return Err(Error::Status(response.status().as_u16()));
-    }
-
-    Ok(response)
-}
-
-// Sends a "request", awaits "response", and parses the body
-// into a deserializable type.
-async fn send_and_parse_response<T: DeserializeOwned>(
-    request: reqwest::RequestBuilder,
-) -> Result<T, Error> {
-    send_and_check_ok(request).await?.json().await.map_err(|e| e.into())
-}
-
-impl Client {
-    pub fn new(address: SocketAddr, log: Logger) -> Client {
-        Client {
-            client: reqwest::Client::new(),
-            log: log.new(o!("propolis_client address" => address.to_string())),
-            address,
+#[cfg(feature = "generated")]
+pub mod generated;
+pub mod handmade;
```

Comment:
From a docs standpoint, do we want the modules themselves to be visible, or simply exposed via the `pub use` below?

Reply:
Added some more clarifying documentation around it and only made `handmade` `pub` when it's not globally `pub use`'d